src/squirrelhooks.ts

```diff
@@ -35,44 +34,16 @@ function runUpdateExe(args: string[]): Promise<void> {
 
 function checkSquirrelHooks(): boolean {
     if (process.platform !== 'win32') return false;
-
     const cmd = process.argv[1];
     const target = path.basename(process.execPath);
-    if (cmd === '--squirrel-install' || cmd === '--squirrel-updated') {
-        runUpdateExe(['--createShortcut=' + target]).then(() => {
-            // remove the old 'Riot' shortcuts, if they exist (update.exe --removeShortcut doesn't work
-            // because it always uses the name of the product as the name of the shortcut: the only variable
-            // is what executable you're linking to)
-            const appDataDir = process.env.APPDATA;
-            if (!appDataDir) return;
-            const startMenuDir = path.join(
-                appDataDir, 'Microsoft', 'Windows', 'Start Menu', 'Programs', 'New Vector Ltd',
-            );
-            return fsProm.rmdir(startMenuDir, { recursive: true });
-        }).then(() => {
-            // same for 'Element (Riot) which is old now too (we have to try to delete both because
-            // we don't know what version we're updating from, but of course we do know this version
-            // is 'Element' so the two old ones are all safe to delete).
-            const appDataDir = process.env.APPDATA;
-            if (!appDataDir) return;
-            const oldStartMenuLink = path.join(
-                appDataDir, 'Microsoft', 'Windows', 'Start Menu', 'Programs', 'Element', 'Element (Riot).lnk',
-            );
-            return fsProm.unlink(oldStartMenuLink).catch(() => {});
-        }).then(() => {
-            const oldDesktopShortcut = path.join(app.getPath('desktop'), 'Element (Riot).lnk');
-            return fsProm.unlink(oldDesktopShortcut).catch(() => {});
-        }).then(() => {
-            const oldDesktopShortcut = path.join(app.getPath('desktop'), 'Riot.lnk');
-            return fsProm.unlink(oldDesktopShortcut).catch(() => {});
-        }).then(() => {
-            app.quit();
-        });
+    if (cmd === '--squirrel-install') {
+        runUpdateExe(['--createShortcut=' + target]).then(() => app.quit());
+        return true;
+    } else if (cmd === '--squirrel-updated') {
+        app.quit();
```

Comment:
I think that the shortcut does need updating here, to re-point to the updated paths.

![image](https://user-images.githubusercontent.com/2403652/162702925-ff76059c-09e4-406a-9efb-e6119fd50196.png)

The `start in` attribute points directly to the updated folder

Reply:
Even with a wrong start in attribute, the app is still launchable and works as intended.

Reply:
I think leaving invalid shortcuts around which could stop working one day is far from ideal and a regression that this PR would cause.

Reply:
I think you're mistaken. If someone makes a change in the future without considering all cases, the regression is on that future PR, not a past PR. The only way I see a regression happening is if someone decided yet again to rename the app name. And talking about regressions, I had to manually download and install Element today because the app wouldn't startup. Now that is far from ideal. My solution is good considering a) the auto-update design itself is far from ideal b) manually checking if a shortcut exists and then running the shortcut updater is verbose. If someone else is with you, then I'll consider doing a shortcut checker; but if we're doing a shortcut checker, you could argue that it too will cause a regression because of a name change.

Reply:
I wrote an email to microsoft, so let's see what they say since their https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-shllink/16cb4ca1-9339-4d0c-a68d-bf1d6cc0f943?redirectedfrom=MSDN specs don't say a thing about behavior.

Reply:
@t3chguy here is the response I got.
> Also… While we can’t comment (and don’t know) whether this Windows behavior will change in the future, I doubt we would make a change that would cause links with bad working directories to stop working where they worked before. As a general rule, Windows strives to maintain compatibility where possible. 

\- Jeff McCashland <jeffm at microsoft dot com>